src/common/options.rs

```diff
@@ -112,13 +109,43 @@ impl Endpointing {
     #[allow(missing_docs)]
     pub fn as_str(&self) -> String {
         match self {
-            Endpointing::True(_value) => "true".to_string(),
-            Endpointing::False(_value) => "false".to_string(),
+            Endpointing::Enabled(value) => {
+                if *value {
+                    "true".to_string()
+                } else {
+                    "false".to_string()
+                }
+            },
             Endpointing::CustomValue(value) => format!("{}", value),
         }
     }
 }
 
+#[derive(Debug, PartialEq, Clone)]
+struct Extra {
```

Comment:
As the user can specify an unbounded number of key value pairs, it's probably best to allow multiple. Instead of a `Vec<Extra>`, we should use the native `HashMap<String,String> type (in `std::collections::HashMap`).